qiskit_nature/problems/second_quantization/electronic/electronic_structure_problem.py

```diff
@@ -175,6 +175,28 @@ def get_default_filter_criterion(
 
         In the fermionic case the default filter ensures that the number of particles is being
         preserved.
+
+        .. note::
```

Comment:
I think this docstring is good now. However, I am not yet sure about its location. I do agree, that this is related to the `ElectronicStructureProblem`, but maybe this should be placed on the class docstring (instead of this method one) to give it a bit more visibility.
The example should also show how this is used in combination with the `NumPyEigensolver`